tests/integration/pkg/tests/custom/custom_operations_test.go

```diff
@@ -0,0 +1,475 @@
+			Name: "test-custom-etcd-snapshot-operations",
+		},
+		Spec: provisioningv1.ClusterSpec{
+			RKEConfig: &provisioningv1.RKEConfig{},
+		},
+	})
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	command, err := cluster.CustomCommand(clients, c)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	assert.NotEmpty(t, command)
+
+	_, err = systemdnode.New(clients, c.Namespace, "#!/usr/bin/env sh\n"+command+" --worker --controlplane", map[string]string{"custom-cluster-name": c.Name})
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	_, err = systemdnode.New(clients, c.Namespace, "#!/usr/bin/env sh\n"+command+" --etcd", map[string]string{"custom-cluster-name": c.Name})
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	_, err = cluster.WaitForCreate(clients, c)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	wContext, err := wrangler.NewContext(context.TODO(), clients.ClientConfig, clients.RESTConfig)
+	if err != nil {
+		t.Fatal(err)
+	}
+	// Register settings so that the provider is set and we can retrieve the internal server URL + CA for the kubeconfig manager below.
+	if err := settings.Register(wContext.Mgmt.Setting()); err != nil {
+		t.Fatal(err)
+	}
+	kcManager := kubeconfig.New(wContext)
+
+	// Get kubeconfig for the downstream cluster to create test resources
+	restConfig, err := kcManager.GetRESTConfig(c, c.Status)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	clientset, err := kubernetes.NewForConfig(restConfig)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	// Create a configmap called "myspecialconfigmap" that we will delete after taking a snapshot.
+	_, err = clientset.CoreV1().ConfigMaps("default").Create(context.TODO(), &corev1.ConfigMap{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: "myspecialconfigmap",
+		},
+		Data: map[string]string{
+			"test": "wow",
+		},
+	}, metav1.CreateOptions{})
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	// Create an etcd snapshot
+	if err := retry.RetryOnConflict(retry.DefaultRetry, func() error {
+		newC, err := clients.Provisioning.Cluster().Get(c.Namespace, c.Name, metav1.GetOptions{})
+		if err != nil {
+			return err
+		}
+		newC.Spec.RKEConfig.ETCDSnapshotCreate = &rkev1.ETCDSnapshotCreate{
+			Generation: 1,
+		}
+		newC, err = clients.Provisioning.Cluster().Update(newC)
+		if err != nil {
+			return err
+		}
+		c = newC
+		return nil
+	}); err != nil {
+		t.Fatal(err)
+	}
+
+	_, err = cluster.WaitForControlPlane(clients, c, "etcd snapshot creation", func(rkeControlPlane *rkev1.RKEControlPlane) (bool, error) {
+		return rkeControlPlane.Status.ETCDSnapshotCreatePhase == rkev1.ETCDSnapshotPhaseFinished, nil
+	})
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	var snapshot *rkev1.ETCDSnapshot
+	// Get the etcd snapshot object
+	if err := retry.OnError(retry.DefaultRetry, func(err error) bool {
+		if apierrors.IsNotFound(err) || err == nil {
+			return true
+		}
+		return false
+	},
+		func() error {
+			snapshots, err := clients.RKE.ETCDSnapshot().List(c.Namespace, metav1.ListOptions{})
+			if err != nil || len(snapshots.Items) == 0 {
+				return err
+			}
```

Comment:
Not sure if it's important but should this return an error if `len(snapshots.Items) == 0`? I figure we want to retry if that slice is empty, but I don't think will if `err` is `nil`. Might have to do something like

```go
if err != nil {
    return err
} else if len(snapshots.Items) == 0 {
    return errors.New("no items found in snapshot")
}
```